### ISSUE TITLE
Lint default param/property values and `@default`/`@defaultvalue`

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#473]
Although the default values as used on jsdoc.app do not necessarily point to valid JavaScript as a default value, e.g., "somebody=John Doe" at https://jsdoc.app/tags-param.html , I favor using standard JavaScript, e.g.,

```js
/**
* @param [aName="a default string"] The param description starts here
* @param [aName=new Map()] Another param description
*/
```

Likewise for `@property` (the [docs](https://jsdoc.app/tags-property.html) don't show defaults as possible, but they do show optional properties, and it makes sense to be present, especially when `@property` accompanies `@typedef`).

I think one should be able to enforce that valid JavaScript is used for a default value, and can be linted (even if, as with `@example`, one may use `overrides` to be able to provide specific rules, e.g., not reporting `no-unused-expressions`).

This rule could also lint `@default`/`@defaultvalue` when a value was present.
